### PR TITLE
Dynamic generation of tilematrixset in GetCapabilities

### PR DESCRIPTION
**Pull request by nadine-piveteau**
To solve issue #2637 and #2636 
Tilesetmatrix are generated automatically for all SRIDs. 
Unittest has been accordingly modified.

[Testlink ](https://mf-chsdi3.dev.bgdi.ch/ltpin_getCap/rest/services/api/1.0.0/WMTSCapabilities.xml?epsg=4326) for EPSG:4236

### CODE REVIEW

**Comment by oterral** [comment]
Funny just saw we have a strange  ServiceMetadatURL at the end of the GetCap:
```
<ServiceMetadataURL xlink:href="http://www.opengis.uab.es/SITiled/world/1.0.0/WMTSCapabilities.xml"/>
```
It's not related to this PR, it's already on producton.

**Comment by oterral** [comment]
@nadine-piveteau trst link doesn't work anymore, deploy a branch please. `make deploybranch`

File: chsdi/templates/wmtscapabilities/wmtscapabilities.mako [hunk 1]
```diff
@@ -7,8 +7,10 @@
   onlineressource = pageargs['onlineressource']
   tilematrixset = pageargs['tilematrixset']
   tmsDefs = pageargs['tilematrixsetDefs']
+  zoomLevels = pageargs['zoomlevels']
   epsg = tilematrixset
   TileMatrixSet_epsg = "TileMatrixSet_%s.mako" % epsg
+  topLeftCorners = {'2056':[2420000.0, 1350000.0], '21781':[420000.0, 350000.0], '3857':[-20037508.3428, 20037508.3428], '4326':[90, -180]}
```
Comment (loicgasser) [inline chsdi/templates/wmtscapabilities/wmtscapabilities.mako]: You should pass in the correct topLeftCorner directly from the view.

File: chsdi/views/wmtscapabilities.py [hunk 2]
```diff
@@ -26,6 +26,15 @@ def getDefaultTileMatrixSet(tileMatrixSet):
     return tilematrixSet
```
Comment (loicgasser) [inline chsdi/views/wmtscapabilities.py]: You could return the topLeftCorner directly from here. gargrid object has a `MAXX`, ` MAXY` attributes.

**Comment by nadine-piveteau** [comment]
Here is the [testlink](http://mf-chsdi3.dev.bgdi.ch/ltpin_getCap/rest/services/api/1.0.0/WMTSCapabilities.xml?epsg=4326).

**Comment by loicgasser** [comment]
`4326_11`
We have
ch.swisstopo.fixpunkte-hfp1
ch.swisstopo.fixpunkte-hfp2
ch.swisstopo.fixpunkte-lfp1
ch.swisstopo.fixpunkte-lfp2

@ltclm seems weird to me that we can't zoom more on these layers.
In geoadmin these layers are WMS. To me this incorrect but I may be wrong.

I tested and it seems to work well thanks.



`

File: chsdi/views/wmtscapabilities.py (hunk 2 above)
Comment (nadine-piveteau) [reply chsdi/views/wmtscapabilities.py]: Ok nice! Shouldn't it be `MAXY` and `MINX`?

Comment (loicgasser) [reply chsdi/views/wmtscapabilities.py]: yes :D

**Comment by ltclm** [comment]
good find @loicgasser 
It is correct to have wms  layers in map.geo.admin.ch because they're updated daily.
I think we should remove them from the wmts getcap, i think it is better to have no data than outdated/wrong data. 
@mariokeusen @gjn what do you think?

**Comment by loicgasser** [comment]
Yes if we don t manage cache update at WMTS level there s no point. +1

**Comment by loicgasser** [comment]
You won't merge this for today's deploy even if it already looks very good.

**Comment by loicgasser** [comment]
```
ch.swisstopo.fixpunkte-hfp1 | 193
ch.swisstopo.fixpunkte-hfp2 | 122
ch.swisstopo.fixpunkte-lfp1 | 118
ch.swisstopo.fixpunkte-lfp2 | 100
```

These are the number of WMTS requests for these layer for the past 7 days.

All of which have been generated either in the tests either by me testing those out, so I removed these from the GetCap.

**Comment by loicgasser** [comment]
@nadine-piveteau is gona have a look at the suspisous ones.

**Comment by ltclm** [comment]
thanks @loicgasser  for removing them

**Comment by gjn** [comment]
If in wmts, they should be managed accordingly. Meaning when data is updated, caches (S3, CF) need to be invalidated.

I think in the future we will have use cases of often updated WMTS layers, automatically updated WMTS layers, etc. That's why the ToD maintenance services are so central. They need to be stable and they need to scale well to enable us in this direction.

For now, I would remove those layers from the WMTS.

**Comment by loicgasser** [comment]
LGTM, please remove the templates we don't use anymore.

**Pull request merged**